force-app/main/default/classes/FieldSetService.cls

```diff
@@ -314,23 +314,25 @@ public with sharing class FieldSetService {
     }
 
     /**
-     * Platform limits the number of reference fields on an object so using this query
-     * within a loop of fields with a field set.
+     * Returns the API name of the "name" field of the object type referenced
+     * by this field.  If the field provided is not a lookup field, this method
+     * returns null.  If the lookup is polymorphic, then the name field for
+     * only one of the referenced types is returned.
      */
-    private String getNameFieldForObject(String sObjectApiName) {
-        if (sObjectApiName == null) {
-            return null;
+    private String getNameFieldForReferenceField(DescribeFieldResult field) {
+        List<Schema.SObjectType> references = field.getReferenceTo();
+        if (!references.isEmpty()) {
```

Comment:
@dospromptman I like to reduce the nesting level and typically would suggest an early return here. Thoughts?